tests/main/snap-quota-journal/task.yaml

```diff
@@ -0,0 +1,96 @@
+
+execute: |
+  echo "Starting service and verifying that log messages are sent to default journal namespace"
+  snap start test-snapd-journal-quota.logger
+
+  # Now ask for logs and lets see if we can see messages from 'LogDaemon'
+  # The format of the log messages are 'LogDaemon: message number: X'
+  retry -n 5 sh -c "$TESTSTOOLS/journal-state get-log | MATCH LogDaemon"
+
+  # Create a journal quota group and set arbitrary limits, we will then verify these are written
+  # correctly for journald
+  echo "Create a group with the logger snap in it"
+  snap set-quota group-one --journal-size=16MB --journal-rate-limit=100/10ms test-snapd-journal-quota
+
+  echo "The systemd slice should be active now"
+  sliceName="snap.$(systemd-escape --path group-one).slice"
+  systemctl show --property=ActiveState "$sliceName" | MATCH "ActiveState=active"
+
+  echo "The service should also still be active"
+  snap services test-snapd-journal-quota.logger | MATCH "test-snapd-journal-quota.logger\s+enabled\s+active"
+
+  # Verify the contents of /etc/systemd/journald@snap-group-one.conf
+  echo "Verifying the journal config contents"
+  MATCH "^SystemMaxUse=16000000"        < /etc/systemd/journald@snap-group-one.conf
+  MATCH "^RuntimeMaxUse=16000000"       < /etc/systemd/journald@snap-group-one.conf
+  MATCH "^RateLimitIntervalSec=10000us" < /etc/systemd/journald@snap-group-one.conf
+  MATCH "^RateLimitBurst=100"           < /etc/systemd/journald@snap-group-one.conf
+  
+  # Now ask for the namespace logs and lets see if we can see messages from 'LogDaemon'
+  retry -n 5 sh -c "$TESTSTOOLS/journal-state get-log --namespace snap-group-one | MATCH LogDaemon"
+
+  # Change the size and ensure that the journal is restarted
+  echo "Updating size of journal to ensure size changes"
+  snap set-quota group-one --journal-size=64MB
+
+  # Ensure that the new size is printed in the journal logs. We requested 64MB but
+  # this will most likely print out as 61.0M, so lets be a bit flexible
+  "$TESTSTOOLS"/journal-state get-log --namespace snap-group-one | MATCH "max 6[0-9].[0-9]M,"
+
+  # Ensure that the unit file is correct
+  echo "Ensuring the unit file is correct"
+  systemctl show --property=LogNamespace snap.test-snapd-journal-quota.logger.service | MATCH "LogNamespace=snap-group-one"
+  LOGGER_PID=$(systemctl show --property=MainPID --value snap.test-snapd-journal-quota.logger.service)
+
+  echo "Removing the quota will stop the slice and the service will be restarted"
+  snap remove-quota group-one
+  systemctl show --property=MainPID --value snap.test-snapd-journal-quota.logger.service | NOMATCH "$LOGGER_PID"
+  snap services test-snapd-journal-quota.logger | MATCH "test-snapd-journal-quota.logger\s+enabled\s+active"
+
+  echo "And the service is not in a slice anymore"
+  systemctl show --property=ControlGroup snap.test-snapd-journal-quota.logger.service | NOMATCH "/$sliceName/snap.test-snapd-journal-quota.logger.service"
```

Comment:
Just to rule out an error in the regular expression, we should show that the same regular expression worked when the slice was active. Something like:
```
SLICE_REGEXP='/$sliceName/snap.test-snapd-journal-quota.logger.service'
# ...before...
systemctl show --property=ControlGroup snap.test-snapd-journal-quota.logger.service | MATCH "$SLICE_REGEXP"
# ...after...
systemctl show --property=ControlGroup snap.test-snapd-journal-quota.logger.service | NOMATCH "$SLICE_REGEXP"
```

Reply:
This is a followup that @Meulengracht should look at when he is back. I leave a TODO there for now.

Reply:
Nevermind, changing from 1s sleep to 0.2s sleep broke this, when not going this everything is fine, sorry for the false alarm.

Reply:
Fixed, thanks for your input @mardy !